corehq/apps/es/case_search.py

```diff
@@ -141,21 +141,32 @@ def case_property_filter(case_property_name, value):
     )
 
 
-def case_property_query(case_property_name, value, fuzzy=False):
+def case_property_query(case_property_name, value, fuzzy=False, mode=None):
     """
     Search for all cases where case property with name `case_property_name`` has text value `value`
     """
     if value is None:
         raise TypeError("You cannot pass 'None' as a case property value")
+    if mode not in ['AND', 'OR', None]:
```

Comment:
if you make these lower case (maybe use constants too) you could just pass it through to `case_property_text_query`